tests/resources/test_file_link.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+from uuid import uuid4
+from mock import patch, Mock
+from botocore.exceptions import ClientError
+
+from citrine.resources.file_link import FileCollection, FileLink, _Uploader
+from tests.utils.session import FakeSession
+from tests.utils.factories import FileLinkDataFactory
+
+
+@pytest.fixture
+def session() -> FakeSession:
+    return FakeSession()
+
+
+@pytest.fixture
+def collection(session) -> FileCollection:
+    return FileCollection(
+        project_id=uuid4(),
+        dataset_id=uuid4(),
+        session=session
+    )
+
+
+@pytest.fixture
+def valid_data() -> dict:
+    return FileLinkDataFactory(url='www.citrine.io', filename='materials.txt')
+
+
+def test_build_equivalence(collection, valid_data):
+    """Test that build() works the same whether called from FileLink or FileCollection."""
+    assert collection.build(valid_data).dump() == FileLink.build(valid_data).dump()
+
+
+def test_string_representation(valid_data):
+    """Test the string representation."""
+    assert str(FileLink.build(valid_data)) == '<File link \'materials.txt\'>'
+
+
+@pytest.fixture
+def uploader() -> _Uploader:
+    """An _Uploader object with all of its fields filled in."""
+    uploader = _Uploader()
```

Comment:
This could be a factory boy factory.

Reply:
Done. Thanks for this and the other suggestions.